addons/website/static/src/snippets/s_table_of_content/000.js

```diff
@@ -85,6 +86,23 @@ const TableOfContent = publicWidget.Widget.extend({
             this.previousPosition = position;
         }
     },
+    /**
+     * @private
+     */
+    _removeInvisibleElements() {
+        const navbar = this.$target[0].querySelector('.s_table_of_content_navbar');
+        const invisibleElements = navbar.querySelectorAll('.table_of_content_link[data-visibility-id]');
```

Comment:
We could even omit the `[data-visibility-id]` part, any hidden entry will be nice to remove, regardless if its hidden because of the visibility feature.